integration/setup.go

```diff
@@ -73,26 +72,28 @@ func newTestSetup(
 	configs []consumerServiceConfig,
 ) *setup {
 	log.SimpleLogger.Debugf("setting up a test with %d producers", numProducers)
-	store := mem.NewStore()
+
 	configService := client.NewMockClient(ctrl)
-	configService.EXPECT().Store(gomock.Any()).Return(store, nil)
-	ts, err := topic.NewService(topic.NewServiceOptions().SetConfigService(configService))
-	require.NoError(t, err)
+	configService.EXPECT().Store(gomock.Any()).Return(mem.NewStore(), nil).AnyTimes()
```

Comment:
Hm feels a bit strange to have mocks in integration tests? Can you use a no-op/in-memory implementation instead?

Reply:
I think doing a no-op/in-memory impl is essentially another way of `mock` so I decided to just use gomock here which is more manageable then in-memory impl

Reply:
Hm ok, not sure how the integration tests are written, hopefully they are done in a way such that the components in the library are treated as black boxes and you are only mocking the callers of the library rather the library components themselves. If that's the case I'm fine.

Reply:
yeah everything in config service are considered as external/black box to m3msg, which is why they are mocked out